src/pyscaffold/shell.py

```diff
@@ -86,8 +88,11 @@ def run(self, *args, **kwargs) -> subprocess.CompletedProcess:
             "universal_newlines": True,
             **kwargs,  # allow overwriting defaults
         }
-        return subprocess.run(command, **opts)
-        # ^ `check_output` does not seem to support terminal editors
+        if self._shell:
+            return subprocess.run(command, **opts)
+            # ^ `check_output` does not seem to support terminal editors
+        return subprocess.run(shlex.split(command, posix=IS_POSIX), **opts)
```

Comment:
This might have been an untested error... I was facing very cryptic errors.

I think originally the class was not thought as receiving a multi-word command string such as `ShellCommand("code --wait")`, but then we changed that for the EDITOR env variable... That might have been the point where things became more complicated.

Reply:
It's funny that after all those years, there are still problems coming up due to calling shell commands. Calling external commands like this is always a minefield and should thus be avoided, at least that's what I keep telling our junior developers. In our case sadly there is no other way as Git has no real library implementation :-(

Reply:
Hopefully it works fine now!
(I think this internal inconsistency was just introduced accidentally when we started accepting "multi-word" commands for the `$EDITOR`, before that I am sure it was fine...)

Reply:
Thanks :-)